### DISCUSSION TITLE
A few missing Latitudes and Longitudes from EIA860 output tables

### DISCUSSION THREAD

**Discussion by ktehranchi** [catalyst-cooperative/pudl#3667] [Help Me!]
Hi there-

I'm noticing a small set of generators are missing latitude and longitude data from the EIA860 plants sheet. See the attached file, notebook, and pudl SQL query below. I am finding 267 generators in the CONUS NERC regions missing lat/long. I can manually match 47 of those with GPS locations from the 2022 EIA plants sheet- so still some 200+ plant locations missing. I couldn't find any trend in operating date, plant type, or plant location (state/ba).

Do you all have any insight on the reason for the missing plants locations? 
[missing_gps_pudl.csv](https://github.com/user-attachments/files/15809050/missing_gps_pudl.csv)

Notebook: https://github.com/ktehranchi/pypsa-usa/blob/backcasting/workflow/notebooks/missing_gps_pudl.ipynb

OG SQL Querry: https://github.com/ktehranchi/pypsa-usa/blob/008baf5514332889beea46e631551b63a3df940f/workflow/scripts/build_powerplants.py#L18

Appreciate your thoughts!

**Comment by zaneselvans** [comment]
This is likely due to inconsistently reported historical lat/lon values. We look at all reported values across all years (and tables, if they show up in multiple places) and if there's no internally consistent value the canonical value (in the plants entity table in this case) ends up being NULL.  There are definitely a handful of plants where obviously incorrect lat/lon values have shown up in the past (zero lat or lon, or at the north pole...) and then there's the issue of using different numbers of sigfigs in the float values.  Right now IIRC we round to some low number of digits and compare the values.  Really we should probably be doing something that treats the lat/lon as a geopoint and uses the average value as long as they're all close to each other after dropping crazy outliers.

This is currently happening in [pudl.transform.eia._lat_long()](https://github.com/catalyst-cooperative/pudl/blob/main/src/pudl/transform/eia.py#L183)

**Comment by ktehranchi** [comment]
Okay interesting, thanks for the quick response. 

I feel like the Geopoint route would be nice. Also, check that the lat-lon exists in the BA/service territory it is listed in. I'll make a note to revisit this, and I would be down to contribute here.
